compiler-cli/src/lsp.rs

```diff
@@ -236,6 +239,12 @@ impl LanguageServer {
         Ok(())
     }
 
+    fn did_open(&mut self) -> Result<()> {
+        // Recompile the project to detect any errors
+        let _ = self.compiler.compile()?;
+        Ok(())
+    }
```

Comment:
Do we need to recompile a project when a file is opened? There will be no changes.

Reply:
At the moment, errors/warnings are not shown when the file is opened

Reply:
I don't wish to compile the project when a file is opened, and this branch is about diagnostics only. Logic around when we compile the project comes with https://github.com/gleam-lang/gleam/issues/1513

Reply:
Awesome, so I will remove this